### ISSUE TITLE
Different handling of added tokens between fast and slow LlamaTokenizer

### ISSUE THREAD

**Issue by cg123** [huggingface/transformers#27279]
### System Info

- `transformers` version: 4.35.0
- Platform: Linux-5.15.0-67-generic-x86_64-with-glibc2.31
- Python version: 3.10.11
- Huggingface_hub version: 0.17.3
- Safetensors version: 0.3.3
- Accelerate version: 0.23.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```python
tok_slow = transformers.AutoTokenizer.from_pretrained("mistralai/Mistral-7B-v0.1", use_fast=False)
tok_slow.add_tokens(["<|special|>", "<|honk|>"])

>>> tok_slow("<|special|>test<|honk|><|special|>\n<|honk|>")
{'input_ids': [1, 32000, 1369, 32001, 32000, 28705, 13, 32001], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1]}

tok_fast = transformers.AutoTokenizer.from_pretrained("mistralai/Mistral-7B-v0.1", use_fast=True)
tok_fast.add_tokens(["<|special|>", "<|honk|>"])
>>> tok_fast("<|special|>test<|honk|><|special|>\n<|honk|>")
{'input_ids': [1, 32000, 1613, 28789, 28766, 21133, 28729, 28766, 3409, 28766, 14908, 28766, 28767, 13, 28789, 28766, 21133, 28729, 28766, 28767], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
```

### Expected behavior

I would expect the tokenizers to handle the added tokens in the same way regardless of use_fast. The behavior of the slow tokenizer is what I was expecting.

**Comment by geronimi73** [comment]
guess it's the same issue as this one: https://github.com/huggingface/transformers/issues/27132

**Comment by ArthurZucker** [comment]
Yep looks pretty similar. This nevered worked for any tokenizers before 😉

**Comment by geronimi73** [comment]
so if I understand the answer in the other thread correctly, using `normalized=False` would yield consistent results for fast and slow tokenizer?

like this for @cg123's case:  
```
tokenizer.add_tokens(AddedToken("<|special|>", normalized=False))
tokenizer.add_tokens(AddedToken("<|honk|>", normalized=False))
```

**Comment by ArthurZucker** [comment]
Yes, if you are using `transformers>=4.34` and `tokenizers>=0.14.0`
